pkg/cmd/certregenerationcontroller/cabundlesyncer.go

```diff
@@ -0,0 +1,149 @@
+package certregenerationcontroller
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/targetconfigcontroller"
+)
+
+const workQueueKey = "key"
+
+type CABundleController struct {
+	kubeClient kubernetes.Interface
+
+	configMapLister corev1listers.ConfigMapLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewCABundleController(
+	kubeClient kubernetes.Interface,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+) (*CABundleController, error) {
+	c := &CABundleController{
+		kubeClient:      kubeClient,
+		configMapLister: kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Lister(),
+		eventRecorder:   eventRecorder.WithComponentSuffix("manage-client-ca-bundle-recovery-controller"),
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "CABundleRecoveryController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Informer().HasSynced)
+
+	handler := cache.ResourceEventHandlerFuncs{
```

Comment:
nit: we usually extract this into func, but this is fine I guess (consistency)

Reply:
I wonder about the reasoning for creating separate lambdas for every informer having the same implementation? The function is not used anywhere else and I only see the informer using the callbacks, not modifying it.